scene/main/node.cpp

```diff
@@ -3027,14 +3028,15 @@ void Node::replace_by(Node *p_node, bool p_keep_groups) {
 
 	_replace_connections_target(p_node);
 
-	if (data.owner) {
-		for (int i = 0; i < get_child_count(); i++) {
-			find_owned_by(data.owner, get_child(i), &owned_by_owner);
-		}
+	if (p_keep_children) {
+		if (data.owner) {
+			for (int i = 0; i < get_child_count(); i++) {
+				find_owned_by(data.owner, get_child(i), &owned_by_owner);
+			}
 
-		_clean_up_owner();
+			_clean_up_owner();
```

Comment:
I don't think this should be controlled by `p_keep_children`, it's unrelated to that, make sure it works correctly otherwise